### PR TITLE
Contributing guidelines

### PR DESCRIPTION
**Pull request by marcofranssen**
This PR also includes #106. It is a recreate of the old #60 PR which was unable to easily rebase on the main branch due to major changes in the branch.

Consider merging #106 first.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+To discuss ideas and specifications we use [Github
+Discussions](https://github.com/notaryproject/notaryproject/discussions).
+
+## Understanding Notary v2
+
+This project is composed of:
+
+- [notation](https://github.com/notaryproject/notation): The Notary v2 CLI and Docker plugins
+- [notation-go-lib](https://github.com/notaryproject/notation-go-lib): A collection of libraries for supporting Notation sign, verify, push, pull of oci artifacts. Based on Notary V2 standard.
```
Comment (shizhMSFT) [inline CONTRIBUTING.md:45]: I know you copied this from the repo description. However, "push, pull" is out of the scope of the `notation-go-lib` and is removed now.

**Comment by marcofranssen** [comment]
@SteveLasker What about code specific guidelines? the other project doesn't have code right?

**Comment by iamsamirzon** [comment]
> I think the question is do we have the same doc in multiple code repos, which will need to be kept in-sync. Or, do we have it in one location, with references from the various repos. We could use guidelines for code and md. For instance, some markdown files have code snippets.

------
One location for the same content is preferred. If the choice for the location is between notaryproject/notaryproject or notaryproject/roadmap, then I recommend we go with the former.

**Comment by sajayantony** [comment]
Recommend accepting the PR since its intention is good and help anyone contributing to notation at this point.

**Review by sajayantony — APPROVED** [review]

**Review by SteveLasker — APPROVED** [review]
Thanks @marcofranssen
LGTM

**Pull request merged**